[Failed guessing mime type of file] Unable to publish ".png" image to the report.

Hi there ! 

# Description
Based on [the proposed by the DVCorg community MLOps tutorial](https://www.youtube.com/watch?v=9BgIDqAzfuA&list=PL7WG7YrwYcnDBDuCkFbcyjnZQrdskFsBz) I was attaching the calculated and saved `confusion_matrix.png` image to the report. But I continuously obtained the error: 
```bash
Error: Failed guessing mime type of file result/confusion_matrix.png
    at mime_type (/cml/src/utils.js:49:11)
Error: Process completed with exit code 1.

```


### The `.github/workflow/expirement.cml` file content:
```bash
name: experiment-workflow
on: [push]
jobs:
  run:
    runs-on: [ubuntu-latest]
    container: docker://dvcorg/cml-py3:latest
    steps:
      - uses: actions/checkout@v2
      - name: cml_run
        env:
          repo_token: ${{ secrets.GITHUB_TOKEN }}
        run: |

          pip install -r requirements.txt
          python expirement.py

          # CML REPORT
          echo "## Model Metrics" >> report.md
          cat results/metrics.txt >> report.md

          echo "## Confusion Matrix" >> report.md
          cml-publish results/confusion_matrix.png --md >> report.md

          cml-send-comment report.md
```

### The sample of the `expirement.py` file content:
```python
# imports
...

_FOLDER = Path('./results')
_METRICS = Path('metrics.txt')
_CONFUSION_MATRIX = Path('confusion_matrix.png')


shutil.rmtree(_FOLDER, ignore_errors=False)
Path(_FOLDER).mkdir(exist_ok=True)

# example of model output
target = [1, 1, 1, 0, 0, 1]
predicted = [1, 1, 1, 1, 1, 0]


results_dict = {
    'Precision': MetricsFactory.precision(target=target, predicted=predicted),
    'Recall': MetricsFactory.recall(target=target, predicted=predicted),
    'ROC AUC': MetricsFactory.roc_auc(target=target, predicted=predicted),
    'F1 Score': MetricsFactory.f1(target=target, predicted=predicted),
    'Confusion Matrix': MetricsFactory.confusion_matrix(target=target, predicted=predicted)
}

with open(_FOLDER / _METRICS, 'w') as f:
    f.write(str(results_dict))

_confusion_matrix = pd.DataFrame(results_dict['Confusion Matrix'])

plt.figure(figsize=(10, 7))
sn.heatmap(_confusion_matrix, annot=True)
plt.savefig(_FOLDER / _CONFUSION_MATRIX)
```

# Summary
The error denotes an issue with the `confusion_matrix.png` file. In the meantime I can open and validate the image exists and format is not corrupted.  
Could you please give a hint what's going wrong ? What did I miss if so ?

--- a/bin/cml-publish.test.js
+++ b/bin/cml-publish.test.js
@@ -105,4 +105,18 @@ describe('CML e2e', () => {
 
     expect(output.startsWith('https://')).toBe(true);
   });
+
+  test('cml-publish /nonexistent produces file error', async () => {
+    await expect(
+      exec(`echo none | node ./bin/cml-publish.js /nonexistent`)
+    ).rejects.toThrowError(
+      'Path /nonexistent does not exist or is not a readable file'
+    );
+  });
+
+  test('echo invalid | cml-publish produces buffer mime type error', async () => {
+    await expect(
+      exec(`echo invalid | node ./bin/cml-publish.js`)
+    ).rejects.toThrowError('Failed guessing mime type of buffer');
+  });
 });

--- a/bin/cml-publish.test.js
+++ b/bin/cml-publish.test.js
@@ -108,10 +108,8 @@ describe('CML e2e', () => {
 
   test('cml-publish /nonexistent produces file error', async () => {
     await expect(
-      exec(`echo none | node ./bin/cml-publish.js /nonexistent`)
-    ).rejects.toThrowError(
-      'Path /nonexistent does not exist or is not a readable file'
-    );
+      exec('echo none | node ./bin/cml-publish.js /nonexistent')
+    ).rejects.toThrowError('ENOENT: no such file or directory, stat');
   });
 
   test('echo invalid | cml-publish produces buffer mime type error', async () => {

--- a/src/utils.js
+++ b/src/utils.js
@@ -54,7 +54,15 @@ const mime_type = async (opts) => {
 
 const fetch_upload_data = async (opts) => {
   const { path, buffer } = opts;
-
+  
+  if (path) {
+    try {
+      await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
+    } catch (err) {
+      throw new Error(`Path ${path} does not exist or is not a readable file.`);
+    }
+  }
+  
   const mime = await mime_type(opts);
   const data = path ? fs.createReadStream(path) : buffer;
   const size = path ? (await fs.promises.stat(path)).size : buffer.length;

--- a/src/utils.js
+++ b/src/utils.js
@@ -54,7 +54,7 @@ const mime_type = async (opts) => {
 
 const fetch_upload_data = async (opts) => {
   const { path, buffer } = opts;
-  
+
   if (path) {
     try {
       await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
@@ -62,7 +62,7 @@ const fetch_upload_data = async (opts) => {
       throw new Error(`Path ${path} does not exist or is not a readable file.`);
     }
   }
-  
+
   const mime = await mime_type(opts);
   const data = path ? fs.createReadStream(path) : buffer;
   const size = path ? (await fs.promises.stat(path)).size : buffer.length;

--- a/src/utils.js
+++ b/src/utils.js
@@ -55,17 +55,9 @@ const mime_type = async (opts) => {
 const fetch_upload_data = async (opts) => {
   const { path, buffer } = opts;
 
-  if (path) {
-    try {
-      await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
-    } catch (err) {
-      throw new Error(`Path ${path} does not exist or is not a readable file`);
-    }
-  }
-
-  const mime = await mime_type(opts);
-  const data = path ? fs.createReadStream(path) : buffer;
   const size = path ? (await fs.promises.stat(path)).size : buffer.length;
+  const data = path ? fs.createReadStream(path) : buffer;
+  const mime = await mime_type(opts);
 
   return { mime, size, data };
 };

--- a/src/utils.js
+++ b/src/utils.js
@@ -59,7 +59,7 @@ const fetch_upload_data = async (opts) => {
     try {
       await fs.promises.access(path, fs.constants.F_OK | fs.constants.R_OK);
     } catch (err) {
-      throw new Error(`Path ${path} does not exist or is not a readable file.`);
+      throw new Error(`Path ${path} does not exist or is not a readable file`);
     }
   }
 

